app/scripts/lib/multichain-api/wallet-requestPermissions.js

```diff
@@ -63,121 +63,118 @@ async function requestPermissionsImplementation(
   }
 
   const [requestedPermissions] = params;
+  const caip25Permission = requestedPermissions[Caip25EndowmentPermissionName];
   delete requestedPermissions[Caip25EndowmentPermissionName];
 
-  const [_grantedPermissions] = await requestPermissionsForOrigin(
-    requestedPermissions,
-  );
-
-  // caveat values are frozen and must be cloned before modified
-  const grantedPermissions = { ..._grantedPermissions };
-
-  const ethAccountsPermission =
-    grantedPermissions[RestrictedMethods.eth_accounts];
+  const legacyRequestedPermissions = pick(requestedPermissions, [
+    RestrictedMethods.eth_accounts,
+    PermissionNames.permittedChains,
+  ]);
+  delete requestedPermissions[RestrictedMethods.eth_accounts];
+  delete requestedPermissions[PermissionNames.permittedChains];
```

Comment:
probably need to have permittedChains/eth_accounts defaults if one is provided but not the other

Reply:
done